dask/sizeof.py

```diff
@@ -135,44 +135,78 @@ def register_pandas():
     import numpy as np
     import pandas as pd
 
-    def object_size(x):
-        if not len(x):
+    from dask.dataframe._compat import dtype_eq
+
+    def object_size(*xs):
+        if not xs:
+            return 0
+        ncells = sum(len(x) for x in xs)
+        if not ncells:
             return 0
-        sample = np.random.choice(x, size=20, replace=True)
-        sample = list(map(sizeof, sample))
-        return sum(sample) / 20 * len(x)
+
+        # Deduplicate Series of references to the same objects,
+        # e.g. as produced by read_parquet
+        unique_samples = {}
+        for x in xs:
+            sample = np.random.choice(x, size=100, replace=True)
+            for i in sample.tolist():
+                unique_samples[id(i)] = i
+
+        nsamples = 100 * len(xs)
+        sample_nbytes = sum(sizeof(i) for i in unique_samples.values())
+        if len(unique_samples) / nsamples > 0.5:
+            # Less than half of the references are duplicated.
+            # Assume that, if we were to analyze twice the amount of random references,
+            # we would get twice the amount of unique objects too.
+            return int(sample_nbytes * ncells / nsamples)
+        else:
+            # Assume we've already found all unique objects and that all references that
+            # we have not yet analyzed are going to point to the same data.
+            return sample_nbytes
```

Comment:
We must be taking up some memory anyway no?  If only to hold onto all of the pointers.  If I have a billion-row series with just "Y" and "N" I'd expect it to take at least a gigabyte.  My understanding here (perhaps flawed) is that we would return less than a GB here in that case.  Is my understanding correct?

Reply:
Ah, we get this by calling `series.memory_usage(deep=False)`.  Is that correct?

Reply:
correct